src/controllers/mainController.ts

```diff
@@ -1288,8 +1288,21 @@ export default class MainController implements vscode.Disposable {
 		this._connectionMgr.onDidOpenTextDocument(doc);
 
 		if (doc && doc.languageId === Constants.languageId) {
+			const configuration = vscode.workspace.getConfiguration(Constants.extensionConfigSectionName);
+			const intelliSenseConfig = configuration.get<IntelliSenseConfig>(Constants.configIntelliSense);
+			const flavor = intelliSenseConfig.defaultQueryEditorLanguageToNone ? Constants.noneProviderName : Constants.mssqlProviderName;
 			// set encoding to false
-			this._statusview.languageFlavorChanged(doc.uri.toString(true), Constants.mssqlProviderName);
+			this._statusview.languageFlavorChanged(doc.uri.toString(true), flavor);
+
+			// Notify the language service that the editor with the specified URI doesn't have a language flavor to avoid error squiggles.
+			if (flavor === Constants.noneProviderName) {
+				SqlToolsServerClient.instance.sendNotification(LanguageFlavorChangedNotification.type,
+					<DidChangeLanguageFlavorParams>{
```

Comment:
nit: we should try to use `{...} as DidChangeLanguageFlavorParams` instead of `<DidChangeLanguageFlavorParams>{...}` to make potential future intermingling with jsx/tsx easier.